typescript/sdk/src/token/types.ts

```diff
@@ -147,7 +151,59 @@ export const WarpRouteDeployConfigSchema = z
     });
 
     return z.NEVER; // Causes schema validation to throw with above issue
+  })
+  // Verify that CCIP hooks are paired with CCIP ISMs
+  .transform((warpRouteDeployConfig, ctx) => {
+    const { ccipHookMap, ccipIsmMap } = getCCIPConfigMaps(
+      warpRouteDeployConfig,
+    );
+
+    // Check hooks have corresponding ISMs
+    const hookConfigHasMissingIsms = Object.entries(ccipHookMap).some(
+      ([originChain, validDestinationChains]) => {
+        if (!ccipIsmMap[originChain]) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `No CCIP ISM found in config for origin chain ${originChain}`,
+          });
+          return true;
+        }
+
+        return Array.from(validDestinationChains).some((chain) => {
+          if (!ccipIsmMap[originChain].has(chain)) {
+            ctx.addIssue({
+              code: z.ZodIssueCode.custom,
+              path: [chain, 'interchainSecurityModule', '...'],
+              message: `Required CCIP ISM not found in config for CCIP Hook with origin chain ${originChain} and destination chain ${chain}`,
+            });
+            return true;
+          }
+          return false;
```

Comment:
a bit confused how this is different from the above case

Reply:
i got caught in two minds with the loops - resolved